### PR TITLE
feat: Added the AMI to machine setup info to runner workflows.

### PR DESCRIPTION
**Pull request by GuptaNavdeep1983**
This PR adds the AMI Id for the runner machine to github workflows.

### CODE REVIEW

File: modules/runners/templates/start-runner.sh [hunk 1]
```diff
@@ -60,6 +62,16 @@ chown -R $run_as .
 echo "Configure GH Runner as user $run_as"
 sudo --preserve-env=RUNNER_ALLOW_RUNASROOT -u "$run_as" -- ./config.sh --unattended --name "$instance_id" --work "_work" $${config}
 
+tee /opt/actions-runner/.setup_info <<EOL
```
Comment (npalm) [inline modules/runners/templates/start-runner.sh:65]: Should the group not have a value?

**Comment by npalm** [comment]
What other values would be nice to set? For example OS or linux distro?

Like the GitHub hosted runners, group OS, group image

**Comment by npalm** [comment]
@GuptaNavdeep1983 can you check the comments please?

File: modules/runners/templates/start-runner.sh (hunk 1 above)
Comment (GuptaNavdeep1983) [reply modules/runners/templates/start-runner.sh:65]: updated.

**Review by npalm — DISMISSED** [review]
LGTM, @ScottGuymer can you have a quick look on the windows scripts?

File: modules/runners/templates/start-runner.ps1 [hunk 2]
```diff
@@ -88,10 +90,18 @@ Invoke-Expression $configCmd
 
 Write-Host "Starting the runner as user $run_as"
 
+$jsonBody = @(
+    @{
+        group='Machine AMI Id'
+        details="$ami_id"
+    }
+)
```
Comment (npalm) [inline modules/runners/templates/start-runner.ps1:98]:
```suggestion
$jsonBody = @(
    @{
        group='Runner Image'
        details="AMI id: $ami_id"
    }
)
```

File: modules/runners/templates/start-runner.sh [hunk 3]
```diff
@@ -60,6 +62,16 @@ chown -R $run_as .
...
+[
+  {
+    "group": "Machine AMI Id",
+    "detail": "$ami_id"
+  }
+]
+EOL
```
Comment (npalm) [inline modules/runners/templates/start-runner.sh:72]: Adding a bit more info, and similar to GH hosted runners
```suggestion
info_arch=$(uname -p)
info_os=$(( lsb_release -ds || cat /etc/*release || uname -om ) 2>/dev/null | head -n1 | cut -d "=" -f2- | tr -d '"')

tee /opt/actions-runner/.setup_info <<EOL
[
  {
    "group": "Operating System",
    "detail": "Distribution: $info_os\nArchitecture: $info_arch"
  },
  {
    "group": "Runner Image",
    "detail": "AMI id: $ami_id"
  }
]
EOL
```

Comment (npalm) [reply modules/runners/templates/start-runner.sh:72]: Example:
image

**Review by npalm — CHANGES_REQUESTED** [review]
Add some small improvements in the suggestions, tested with the default example

File: modules/runners/templates/start-runner.ps1 [hunk 4]
```diff
@@ -88,10 +90,18 @@ Invoke-Expression $configCmd
 
 Write-Host "Starting the runner as user $run_as"
 
+$jsonBody = @(
+    @{
+        group='Runner Image'
+        details="AMI id: $ami_id"
+    }
+)
+ConvertTo-Json -InputObject $jsonBody > "$pwd\.setup_info"
```
Comment (ScottGuymer) [inline modules/runners/templates/start-runner.ps1:99]:
```suggestion
ConvertTo-Json -InputObject $jsonBody | Set-Content -Path "$pwd\.setup_info"
```

Small change to be more explicit for powershell

**Review by npalm — APPROVED** [review]
Thx, nice job!

**Pull request merged**